src/feed.rs

```diff
@@ -599,7 +599,15 @@ impl Feed<RandomAccessDisk> {
     // NOTE: Should we `mkdirp` here?
     // NOTE: Should we call these `data.bitfield` / `data.tree`?
     pub async fn open<P: AsRef<Path>>(path: P) -> Result<Self> {
+        let metadata = std::fs::metadata(&path)?;
+        if !metadata.is_dir() {
+            let context = format!("{} is not a directory", path.as_ref().display(),);
+
+            return Err(anyhow::Error::msg("Not a directory").context(context));
+        }
```

Comment:
With `std::fs` is imported at the top of the file, this can be rewritten more succinctly as:

```suggestion
        if !fs::metadata(&path)?.is_dir() {
            return Err(anyhow::Error::msg(format!("{} is not a directory", path.as_ref()));
        }
```